portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -30,29 +35,58 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // If user is logged in and has not set a nickname, redirect to nickname page
     UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      
+    }
+    
+    // Create variables needed to create a user.
```

Comment:
nit: For clarity, you could change this to "Create variables needed to instantiate a User object."

Reply:
Done!